geth/jail/internal/loop/loop_test.go

```diff
@@ -0,0 +1,56 @@
+package loop_test
```

Comment:
Please, use the same package for unit tests.

Reply:
Happy to change this, but seems inconsistent with the rest of the code base.

Reply:
I tried to change the package to `package loop` but it gives an error: `import cycle not allowed in test`.

Is this something that is necessary to change?

Reply:
This is a bit tricky: `looptask` imports package `loop` because it has implementations of `type Task`. In `loop_test.go`, package `looptask` is imported so changing the package name to `loop` causes a cyclic import (which is strictly forbidden in Go).

Using a different package name in a test file means you're not interested in using internals of package `loop` so it becomes a black box testing file. But for unit testing:
- The package name is usually the same.
- Some of the internals may/should also be proven to work.
- Most importantly, the tests should be independent from the importing packages.

So it might be best to [mock](https://github.com/golang/mock) `type Task` (provides more flexibility) or create a simple `type testTask` in `loop_test.go` and use it in tests. Ideally, in case anything changes in package `looptask`, unit tests in package `loop` shouldn't fail. I hope it makes sense.

Reply:
Thanks for the suggestion. I used mock to get rid of the need for a `looptask` import.